### PR TITLE
Refactor: Combination

### PR DESCRIPTION
**Pull request by samuelfac**
Related to #5164

- [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [X] This pull request is all my own work -- I have not plagiarized it.
- [X] All filenames are in PascalCase.
- [X] All functions and variable names follow Java naming conventions.
- [X] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [X] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/test/java/com/thealgorithms/backtracking/CombinationTest.java [hunk 1]
```diff
@@ -11,21 +12,23 @@ public class CombinationTest {
...
     }
 
     @Test
     void testLengthOne() {
         List<TreeSet<Integer>> result = Combination.combination(new Integer[] {1, 2}, 1);
-        assertTrue(result.get(0).iterator().next() == 1);
-        assertTrue(result.get(1).iterator().next() == 2);
+        assert result != null;
+        assertEquals(1, (int) result.get(0).getFirst());
+        assertEquals(2, (int) result.get(1).getFirst());
```
Comment (vil02) [inline src/test/java/com/thealgorithms/backtracking/CombinationTest.java:23]:
```suggestion
        assertEquals(Arrays.asList(new TreeSet<>(Arrays.asList(1)), new TreeSet<>(Arrays.asList(2))), result);
```

File: src/test/java/com/thealgorithms/backtracking/CombinationTest.java [hunk 2]
```diff
@@ -11,21 +12,23 @@ public class CombinationTest {
...
     @Test
     void testLengthTwo() {
         List<TreeSet<Integer>> result = Combination.combination(new Integer[] {1, 2}, 2);
-        Integer[] arr = result.get(0).toArray(new Integer[2]);
-        assertTrue(arr[0] == 1);
-        assertTrue(arr[1] == 2);
+        assert result != null;
+        Integer[] arr = result.getFirst().toArray(new Integer[2]);
+        assertEquals(1, (int) arr[0]);
+        assertEquals(2, (int) arr[1]);
```
Comment (vil02) [inline src/test/java/com/thealgorithms/backtracking/CombinationTest.java:32]:
```suggestion
        assertEquals(Arrays.asList(new TreeSet<>(Arrays.asList(1, 2))), result);
```

File: src/main/java/com/thealgorithms/backtracking/Combination.java [hunk 3]
```diff
@@ -1,62 +1,58 @@
...
+     * @throws IllegalArgumentException if combinationSize is negative.
      */
-    public static <T> List<TreeSet<T>> combination(T[] arr, int n) {
-        if (n == 0) {
+    public static <T> List<TreeSet<T>> combination(T[] data, int combinationSize) {
+        if (combinationSize < 0) {
+            throw new IllegalArgumentException("Combination size cannot be negative.");
+        } else if (combinationSize == 0) {
             return null;
         }
```
Comment (vil02) [inline src/main/java/com/thealgorithms/backtracking/Combination.java:29]: I do not see a reason to return `null` here. You can just remove this _branch_, so the result would be `Arrays.asList(new TreeSet<Integer>())`.

The problem is `ArrayCombination`. The method [`combination` will need an update](https://github.com/TheAlgorithms/Java/blob/26b4b82949ae2afb4e2aab94a2c96ee07acff5aa/src/main/java/com/thealgorithms/backtracking/ArrayCombination.java#L23).

File: src/test/java/com/thealgorithms/backtracking/CombinationTest.java
Comment (vil02) [inline src/test/java/com/thealgorithms/backtracking/CombinationTest.java:1]: Could you please:
- add a test case with `data` being an empty array and `combinationSize` being positive,
- add a test case with `data` being an empty array and `combinationSize` being zero,
- add a test case with `data` being `[1, 2, 3]` and `combinationSize` being `2`,
- add a test case with data being an array of strings.

File: src/main/java/com/thealgorithms/backtracking/Combination.java [hunk 4]
```diff
@@ -1,62 +1,58 @@
...
-import java.util.Arrays;
-import java.util.LinkedList;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.TreeSet;
 
 /**
- * Finds all permutations of given array
- * @author Alan Piao (<a href="https://github.com/cpiao3">git-Alan Piao</a>)
+ * Finds all combinations of a given array of unique elements.
```
Comment (vil02) [inline src/main/java/com/thealgorithms/backtracking/Combination.java:8]: This is not verified anywhere. Maybe the input type should be `Set<T>`? Or maybe this statement is not needed. I think what we are aiming for it to mimic the [`itertools.combinations`](https://docs.python.org/3/library/itertools.html#itertools.combinations) from python.

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/test/java/com/thealgorithms/backtracking/CombinationTest.java [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package com.thealgorithms.backtracking;
 
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
 
 import java.util.List;
```
Comment (vil02) [inline src/test/java/com/thealgorithms/backtracking/CombinationTest.java:6]: Regarding the failing build, try to:

```suggestion
import java.util.Arrays;
import java.util.List;
```
